frontend/src/components/SystemsTable/index.tsx

```diff
@@ -67,49 +74,71 @@ export function SystemsTable({
       }
     }
     // if a task is selected, add all supported metrics to the options list
-    if (taskFilter) {
-      const task = findTask(taskCategories, taskFilter);
+    if (filters.task) {
+      const task = findTask(taskCategories, filters.task);
       if (task)
         task.supported_metrics.forEach((metric) => metricNames.add(metric));
     }
     return Array.from(metricNames);
   }
   const metricNames = getMetricsNames();
 
+  /* Fetch all tasks from backend during page initialization */
   useEffect(() => {
-    function getInitialSortField(taskCategories: TaskCategory[]) {
-      if (initialTaskFilter) {
-        const initialTask = findTask(taskCategories, initialTaskFilter);
-        if (initialTask && initialTask.supported_metrics.length > 0)
-          return initialTask.supported_metrics[0];
-      }
-      return "created_at";
-    }
     async function fetchTasks() {
       const taskCategoriesData = await backendClient.tasksGet();
       setTaskCategories(taskCategoriesData);
-      setSortField(getInitialSortField(taskCategoriesData));
     }
     fetchTasks();
-  }, [initialTaskFilter]);
+  }, []);
+
+  /** TODO: add debounce */
+  const onFilterChange = useCallback(
```

Comment:
Could you please double-check if this `useCallback` is necessary?

Reply:
Thanks for pointing out! Made it a plain funcion and it worked.